### PR TITLE
Version 0.4.1 + directory for SCAP content

### PR DESCRIPTION
**Pull request by shlomizadok**
A few notes:
As of this version (and 0.3.4 for foreman-1.7), smart_proxy_openscap serves scap files to the clients (aka, foreman_scap_client).
The configuration where to save the scap files is set [here](https://github.com/OpenSCAP/smart_proxy_openscap/blob/master/lib/smart_proxy_openscap/openscap_plugin.rb#L22)

and saving / serving the files is set [here](https://github.com/OpenSCAP/smart_proxy_openscap/blob/master/lib/smart_proxy_openscap/openscap_lib.rb#L34-L35)

Thanks.

### CODE REVIEW

File: rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec [hunk 1]
```diff
@@ -82,6 +83,11 @@ mv %{buildroot}%{gem_instdir}/extra/smart-proxy-openscap-send.cron \
 # create spool directory
 mkdir -p %{buildroot}%{spool_dir}
 
+# create content directory and symlink it to foreman-proxy directory
+mkdir -p %{buildroot}%{content_dir}/content
+%attr(-, %{proxy_user}, %{proxy_user}) %{buildroot}%{content_dir}
```
Comment (domcleal) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: that's for %files, no?

File: rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec [hunk 2]
```diff
@@ -3,10 +3,11 @@
 %global foreman_proxy_bundlerd_dir %{_datadir}/foreman-proxy/bundler.d
 %global foreman_proxy_pluginconf_dir %{_sysconfdir}/foreman-proxy/settings.d
 %global spool_dir %{_var}/spool/foreman-proxy/openscap
+%global content_dir %{_var}/lib/openscap
```
Comment (domcleal) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: aka %{_sharedstatedir}

**Comment by domcleal** [comment]
There's no 0.4.1 published on rubygems.org, that should be done first.

smart_proxy_openscap is not published for 1.7, only 1.8+.

File: rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec (hunk 2 above)
Comment (domcleal) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: shouldn't this be in a directory related to foreman-proxy, or is it meant to refer to another package's location?

**Comment by shlomizadok** [comment]
Just pushed on [rubygems.org](https://rubygems.org/gems/smart_proxy_openscap/versions/0.4.1)
ACK on 1.8+

File: rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec (hunk 2 above)
Comment (shlomizadok) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: It is related to foreman-proxy and I wouldn't mind creating 'openscap/content' directory under foreman-proxy's directory.
Please also note that openscap is using /var/lib/openscap and creates different directories (like xccdf_scans, etc) - so I thought of adding a new directory (content) and save scap xml files over there - but that it not a must.
Waiting your guidance on what is preferred:
/var/lib/openscap/content symlinked to /path/to/proxy/openscap/content
-or-
just create /path/to/proxy/openscap/content
Thanks.

Comment (domcleal) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: Sounds like this should be under /var/lib/foreman-proxy then, if it's files managed by the foreman-proxy process.

File: rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec [hunk 3]
```diff
@@ -90,6 +95,7 @@ mkdir -p %{buildroot}%{spool_dir}
 %config(noreplace) %{foreman_proxy_pluginconf_dir}/openscap.yml
 
 %attr(-,%{proxy_user},%{proxy_user}) %{spool_dir}
+%attr(-, %{proxy_user}, %{proxy_user}) %{buildroot}%{content_dir}
```
Comment (domcleal) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: remove %{buildroot} as %files is relative

File: rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec [hunk 4]
```diff
@@ -82,6 +83,10 @@ mv %{buildroot}%{gem_instdir}/extra/smart-proxy-openscap-send.cron \
 # create spool directory
 mkdir -p %{buildroot}%{spool_dir}
 
+# create content directory and symlink it to foreman-proxy directory
+mkdir -p %{buildroot}%{content_dir}/content
+ln -sv %{buildroot}%{content_dir} %{buildroot}%{_datadir}/foreman-proxy/openscap
```
Comment (domcleal) [inline rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec]: symlink target should not be inside the buildroot, just %{content_dir}

    error: Symlink points to BuildRoot: /usr/share/foreman-proxy/openscap -> /builddir/build/BUILDROOT/rubygem-smart_proxy_openscap-0.4.1-1.git.96.6691a12.git.96.6691a12.git.96.6691a12.el7.noarch/var/lib/foreman-proxy/openscap

**Comment by shlomizadok** [comment]
:dancers: thanks @domcleal :)

**Comment by domcleal** [comment]
Thanks @shlomizadok, building for nightly + 1.8.

**Pull request merged**